analyzer/api.go

```diff
@@ -160,46 +160,53 @@ const (
 
 // String returns the string representation of an Event.
 func (e *Event) String() string {
-	switch *e {
-	case EventStakingTransfer:
-		return "staking.transfer"
-	case EventStakingBurn:
-		return "staking.burn"
-	case EventStakingAddEscrow:
-		return "staking.escrow.add"
-	case EventStakingTakeEscrow:
-		return "staking.escrow.take"
-	case EventStakingDebondingStart:
-		return "staking.escrow.debonding_start"
-	case EventStakingReclaimEscrow:
-		return "staking.escrow.reclaim"
-	case EventStakingAllowanceChange:
-		return "staking.allowance_change"
-	case EventRegistryRuntime:
-		return "registry.runtime"
-	case EventRegistryEntity:
-		return "registry.entity"
-	case EventRegistryNode:
-		return "registry.node"
-	case EventRegistryNodeUnfrozen:
-		return "registry.node_unfrozen"
-	case EventRoothashExecutorCommitted:
-		return "roothash.executor_committed"
-	case EventRoothashExecutionDiscrepancyDetected:
-		return "roothash.execution_discrepancy_detected"
-	case EventRoothashFinalized:
-		return "roothash.finalized"
-	case EventGovernanceProposalSubmitted:
-		return "governance.proposal_submitted"
-	case EventGovernanceProposalExecuted:
-		return "governance.proposal_executed"
-	case EventGovernanceProposalFinalized:
-		return "governance.proposal_finalized"
-	case EventGovernanceVote:
-		return "governance.vote"
-	default:
-		return "unknown"
+	name, ok := EventToString[*e]
+	if ok {
+		return name
 	}
+	return "unknown"
+}
+
+var EventToString map[Event]string = map[Event]string{
```

Comment:
Thanks for the PR! 

I'd still prefer to validate the input params manually, so that we can return a more helpful error message and also just in case someone tries to inject some bad inputs (perhaps the query parameter structure already prevents that?).

Reply:
SGTM. I'm not worried about sql injection, that's what we have pgx for. Helpful error messages are nice though.

Note you'll have to rebuild StringToEvent (now better named StringToEventType) with the new enum.